c7n/resources/dynamodb.py

```diff
@@ -33,3 +36,37 @@ def _augment(resource_set):
         with self.executor_factory(max_workers=3) as w:
             return list(itertools.chain(
                 *w.map(_augment, chunks(resources, 20))))
+
+
+@Table.action_registry.register('delete')
+class DeleteTable(BaseAction):
+    """Action to delete dynamodb tables
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: delete-tables
+                resource: dynamodb-table
+                actions:
+                  - delete
+    """
+
+    schema = type_schema('delete')
+
+    def delete_table(self, resource):
+        print resource.get('TableName')
+        client = local_session(self.manager.session_factory).client('dynamodb')
+        client.delete_table(TableName=resource['TableName'])
+
+    def process(self, resources):
+        for r in resources:
+            with self.executor_factory(max_workers=3) as w:
+                futures = []
+                futures.append(w.submit(self.delete_table, r))
```

Comment:
crossing the thread boundary with concurrent futures is expensive for single api calls, use chunked resources.